In [9]:
#Necessary Imports

import pandas as pd


In [10]:
file_path = r"C:\Users\dekis\OneDrive\Desktop\DATA SCIENCE\FINAL PROJECT\ds-final-project-main\ds-final-project-main\twitter_racism_parsed_dataset.csv"

# Load the dataset into a pandas DataFrame
data = pd.read_csv(file_path)

data

index            id  \
0      5.767493e+17  5.767493e+17   
1      5.408905e+17  5.408905e+17   
2      5.678433e+17  5.678433e+17   
3      5.766462e+17  5.766462e+17   
4      5.713492e+17  5.713492e+17   
...             ...           ...   
13466  5.755964e+17  5.755964e+17   
13467  5.580631e+17  5.580631e+17   
13468  5.755969e+17  5.755969e+17   
13469  5.755990e+17  5.755990e+17   
13470  5.755977e+17  5.755977e+17   

                                                    Text Annotation  oh_label  
0      @AAlwuhaib1977 Muslim mob violence against Hin...     racism         1  
1                 @Te4m_NiGhtM4Re http://t.co/5Ih7MkDbQG       none         0  
2      @jncatron @isra_jourisra @AMPalestine Islamoph...     racism         1  
3      Finally I'm all caught up, and that sudden dea...       none         0  
4                 @carolinesinders @herecomesfran *hugs*       none         0  
...                                                  ...        ...       ...  
13466   #mkr NOOOOO!!! I wanted Kat and Andre to lose!!!       none         0  
13467  RT @MumtazCeltik: @WhiteHouse @VP \n\n#Kobane ...       none         0  
13468  Glad the proper competition is starting and ev...       none         0  
13469  No worries Pete and Manu, take all the time of...       none         0  
13470  @mykitchenrules #mkr please kick off that lady...       none         0  

[13471 rows x 5 columns]

In [3]:

#the link to the dataset
print("Dataset URL: https://huggingface.co/datasets/sakren/twitter_racism_dataset")


Dataset URL: https://huggingface.co/datasets/sakren/twitter_racism_dataset


In [4]:
# the shape of the dataset
print("Dataset shape:", data.shape)


Dataset shape: (13471, 5)


In [5]:
# head of the dataset 
print("First few rows of the dataset:")
print(data.head())

First few rows of the dataset:
          index            id  \
0  5.767493e+17  5.767493e+17   
1  5.408905e+17  5.408905e+17   
2  5.678433e+17  5.678433e+17   
3  5.766462e+17  5.766462e+17   
4  5.713492e+17  5.713492e+17   

                                                Text Annotation  oh_label  
0  @AAlwuhaib1977 Muslim mob violence against Hin...     racism         1  
1             @Te4m_NiGhtM4Re http://t.co/5Ih7MkDbQG       none         0  
2  @jncatron @isra_jourisra @AMPalestine Islamoph...     racism         1  
3  Finally I'm all caught up, and that sudden dea...       none         0  
4             @carolinesinders @herecomesfran *hugs*       none         0  


In [7]:
#summary stats

print(data.describe(include='all'))

               index            id  \
count   1.347100e+04  1.347100e+04   
unique           NaN           NaN   
top              NaN           NaN   
freq             NaN           NaN   
mean    5.645450e+17  5.645450e+17   
std     2.263979e+16  2.263979e+16   
min     3.208178e+17  3.208178e+17   
25%     5.637581e+17  5.637581e+17   
50%     5.714068e+17  5.714068e+17   
75%     5.755013e+17  5.755013e+17   
max     5.922733e+17  5.922733e+17   

                                                     Text Annotation  \
count                                               13471      13471   
unique                                              13471          2   
top     @AAlwuhaib1977 Muslim mob violence against Hin...       none   
freq                                                    1      11501   
mean                                                  NaN        NaN   
std                                                   NaN        NaN   
min                                    

In [8]:

# information about how the data is structured
print("Information about the dataset:")
data.info()

Information about the dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13471 entries, 0 to 13470
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   index       13471 non-null  float64
 1   id          13471 non-null  float64
 2   Text        13471 non-null  object 
 3   Annotation  13471 non-null  object 
 4   oh_label    13471 non-null  int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 526.3+ KB


## Brief summary of the dataset:

The dataset is sourced from Hugging Face "Twitter Racism Dataset," it contains 13,471 tweets, each annotated to indicate whether it includes racist content. Key features of the dataset include a unique tweet ID, the tweet text, and a binary label (oh_label), where '1' indicates racist content and '0' indicates none. This dataset is particularly suited to detect and analyze racist language on social media.